link_checker/check.sh

```diff
@@ -106,7 +106,15 @@ test_link () {
 		return 1 # Missing scheme, www.example.com will be linked to a file in the repo
 
 	else
-		http_code=$(curl --location --silent -o /dev/null -w '%{http_code}' "${link}")
+		http_code="$(
+			curl \
+			--header 'user-agent: sendwithus-link-checker/2022.06.07' \
```

Comment:
I feel like this might never be updated, even when the script is updated next. Is it worth pulling this out to a variable at the top of the file to increase its visibility?

Reply:
As long as this user-agent passes the filters on some sites that have been blocking curl's default of `user-agent: curl/<curl-version>` this doesn't need to be updated when this script changes. To match the normal UA format, I needed a `name/version` string, but I didn't know what our version number should have been, so I opted for a [calver](https://calver.org/) scheme instead of [semver](https://semver.org/).

Happy to refactor it; maybe it is better to bump this every time there is a change to the script, I just didn't think our version number is much of a concern for anyone. Leave a :+1: reaction on this and I'll move it up top for you :)

Reply:
@alannadare The link checker script is being blocked by some sites when it checks if the job postings are valid or not, since it just looks like a generic bot. By adding a custom user-agent, the link checker looks less generic, and the sites allow it to check the links.

Reply:
@zevisert I don't think remembering to update the calver number is too much of a concern, but IMO by providing it, it does sort of imply that it would change when the script changes. I figure we might as well make the var a bit more obvious so that updating it is less likely to be forgotten next time we do change this script.

Reply:
Sure! On every change made to this repository, we run this script. It is responsible for the "build status" of this repository. 
<img width="367" alt="image" src="https://user-images.githubusercontent.com/11222441/176038798-4f61a57d-52ae-4401-aca1-848cbc23da6c.png">

What this whole script tries to do is figure out _if the job postings you are showing are still available_. It does this by looking at the standardized HTTP response code that is sent back from every website that has a link in the README.md file.

Ideally, we get a successful "build" if all of the jobs are still available, and a failed "build" if any of the jobs have been taken down on their respective websites. The HTTP code isn't always the perfect indicator though; for example a job posting could be replaced with a website that says "Sorry, this position has been filled", but the webserver still sends a "everything in working order" response code.

With this change that I'm proposing specifically, I'm suggesting that we make up a name to call ourselves (a "user-agent") when communicating with webservers hosting the jobs. The reason for this is because _some_ servers automatically block any incoming traffic if it has `curl`'s default user-agent. Without a change like this, some links look fine in the browser but then those same links can appear broken when requested with the program we use to get the job page.

| <img width="1182" alt="image" src="https://user-images.githubusercontent.com/11222441/176041974-6140e2a1-674d-422d-9309-a85e884e383f.png"> |
| ----------------------------- |
| _Left_: Browser Default User Agent --  _Center:_  `curl/7.68.0` -- _Right_: `sendwithus-link-checker/2022.06.07` |

Hope this helps!